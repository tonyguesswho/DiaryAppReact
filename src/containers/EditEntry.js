import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import { connect } from 'react-redux';
import { editEntry, getEntry } from '../actions/entriesActions';
import { clearError } from '../actions/authAction';

class EditEntry extends Component {
  constructor() {
    super();
    this.state = {
      title: '',
      textarea: '',
      error: '',
      loading: false,
    };
  }

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };
  onSubmit = e => {
    e.preventDefault();
    const newEntry = {
      title: this.state.title.trim(),
      description: this.state.textarea.trim()
    };
    this.setState({
      loading: true
    });
    const { id } = this.props.match.params;
    this.props.editEntry(newEntry,id, this.props.history);
  };
  componentDidMount = async () => {
    const { clearError, getEntry } = this.props;
    clearError();
    const { id } = this.props.match.params;
    await getEntry(id);
  };

  logout = () => {
    signout();
  };
  componentWillReceiveProps(nextProps) {
    if (nextProps) {
      const { errorMessage, entry } = nextProps;
      this.setState({
        error: errorMessage,
        loading: false,
        title: entry.title,
        textarea: entry.description
      });

    }
  }
  render() {
    const { loading, error } = this.state;
    return (
      <div className="bg body">
        <header className="header2">
          <nav>
            <ul className="nav nav2">
              <li>
                <Link to="/">Home</Link>
              </li>
              <li>
                <Link to="/profile">Profile</Link>
              </li>
              <li>
                <Link to="/" id="logout" onClick={this.logout}>
                  Sign Out
                </Link>
              </li>
            </ul>
          </nav>
        </header>
        <section className="signup" id="messageBox">
          <div className="box span31 span3-center card">
            {error && <div className="msg_output_fail">{error}</div>}
            <h2>Edit Diary Entry </h2>
            <form onSubmit={this.onSubmit} id="addEntry">
              <div className="fullspan">
                <label htmlFor="title">Title:</label>
              </div>
              <div className="fullspan">
                <input type="text" name="title" id="title" placeholder="Entry title"  required onChange={this.onChange} value={this.state.title} />
              </div>
              <div className="fullspan">
                <label htmlFor="textarea">Express Yourself:</label>
              </div>
              <div className="fullspan">
                <textarea name="textarea" id="textarea" rows="10" required onChange={this.onChange} value={this.state.textarea} />
              </div>

              {!loading ? <input type="submit" value="Edit Entry" /> : <input type="submit" value="Loading  . . ." disabled="disabled" />}
            </form>
          </div>
        </section>
      </div>
    );
  }
}

const mapStateToProps = state => ({ entry: state.entries.entry, errorMessage:state.entries.entryError });

export default connect(
  mapStateToProps,
  { editEntry,getEntry, clearError }
)(withRouter(EditEntry));
