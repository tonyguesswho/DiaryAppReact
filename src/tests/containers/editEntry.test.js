import React from 'react';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import toJson from 'enzyme-to-json';
import EditEntry from '../../containers/EditEntry';
import Root from '../../root';

let wrapped;

beforeEach(() => {
  wrapped = mount(
    <MemoryRouter initialEntries={[{ key: 'testKey' }]}>
      <Root>
        <EditEntry />
      </Root>
    </MemoryRouter>
  );
});

afterEach(() => wrapped.unmount());

describe('Edit entry UI', () => {
  describe('render features', () => {
    test('container should render as expected', () => {
      const tree = toJson(wrapped);
      expect(tree).toMatchSnapshot();
    });
  });
});
