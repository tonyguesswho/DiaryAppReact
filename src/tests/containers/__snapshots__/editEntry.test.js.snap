// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Edit entry UI render features container should render as expected 1`] = `
<MemoryRouter
  initialEntries={
    Array [
      Object {
        "key": "testKey",
      },
    ]
  }
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "canGo": [Function],
        "createHref": [Function],
        "entries": Array [
          Object {
            "hash": "",
            "key": "testKey",
            "pathname": "/",
            "search": "",
          },
        ],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "index": 0,
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "key": "testKey",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <_default>
      <Provider
        store={
          Object {
            "dispatch": [Function],
            "getState": [Function],
            "replaceReducer": [Function],
            "subscribe": [Function],
            Symbol(observable): [Function],
          }
        }
      >
        <Connect(withRouter(EditEntry))>
          <withRouter(EditEntry)
            clearError={[Function]}
            editEntry={[Function]}
            entry={Object {}}
            errorMessage=""
            getEntry={[Function]}
          >
            <Route>
              <EditEntry
                clearError={[Function]}
                editEntry={[Function]}
                entry={Object {}}
                errorMessage=""
                getEntry={[Function]}
                history={
                  Object {
                    "action": "POP",
                    "block": [Function],
                    "canGo": [Function],
                    "createHref": [Function],
                    "entries": Array [
                      Object {
                        "hash": "",
                        "key": "testKey",
                        "pathname": "/",
                        "search": "",
                      },
                    ],
                    "go": [Function],
                    "goBack": [Function],
                    "goForward": [Function],
                    "index": 0,
                    "length": 1,
                    "listen": [Function],
                    "location": Object {
                      "hash": "",
                      "key": "testKey",
                      "pathname": "/",
                      "search": "",
                    },
                    "push": [Function],
                    "replace": [Function],
                  }
                }
                location={
                  Object {
                    "hash": "",
                    "key": "testKey",
                    "pathname": "/",
                    "search": "",
                  }
                }
                match={
                  Object {
                    "isExact": true,
                    "params": Object {},
                    "path": "/",
                    "url": "/",
                  }
                }
              >
                <div
                  className="bg body"
                >
                  <header
                    className="header2"
                  >
                    <nav>
                      <ul
                        className="nav nav2"
                      >
                        <li>
                          <Link
                            replace={false}
                            to="/"
                          >
                            <a
                              href="/"
                              onClick={[Function]}
                            >
                              Home
                            </a>
                          </Link>
                        </li>
                        <li>
                          <Link
                            replace={false}
                            to="/profile"
                          >
                            <a
                              href="/profile"
                              onClick={[Function]}
                            >
                              Profile
                            </a>
                          </Link>
                        </li>
                        <li>
                          <Link
                            id="logout"
                            onClick={[Function]}
                            replace={false}
                            to="/"
                          >
                            <a
                              href="/"
                              id="logout"
                              onClick={[Function]}
                            >
                              Sign Out
                            </a>
                          </Link>
                        </li>
                      </ul>
                    </nav>
                  </header>
                  <section
                    className="signup"
                    id="messageBox"
                  >
                    <div
                      className="box span31 span3-center card"
                    >
                      <h2>
                        Edit Diary Entry 
                      </h2>
                      <form
                        id="addEntry"
                        onSubmit={[Function]}
                      >
                        <div
                          className="fullspan"
                        >
                          <label
                            htmlFor="title"
                          >
                            Title:
                          </label>
                        </div>
                        <div
                          className="fullspan"
                        >
                          <input
                            id="title"
                            name="title"
                            onChange={[Function]}
                            placeholder="Entry title"
                            required={true}
                            type="text"
                            value=""
                          />
                        </div>
                        <div
                          className="fullspan"
                        >
                          <label
                            htmlFor="textarea"
                          >
                            Express Yourself:
                          </label>
                        </div>
                        <div
                          className="fullspan"
                        >
                          <textarea
                            id="textarea"
                            name="textarea"
                            onChange={[Function]}
                            required={true}
                            rows="10"
                            value=""
                          />
                        </div>
                        <input
                          type="submit"
                          value="Edit Entry"
                        />
                      </form>
                    </div>
                  </section>
                </div>
              </EditEntry>
            </Route>
          </withRouter(EditEntry)>
        </Connect(withRouter(EditEntry))>
      </Provider>
    </_default>
  </Router>
</MemoryRouter>
`;
